src/FilePreviewer.js

```diff
@@ -26,206 +51,208 @@ import ViewportContent from './ViewportContent';
-        const promises = R.times(async i => {
-            // Wait until the original sizes gets calculated.
-            const page = await pdf.getPage(i + 1);
-            return R.pick(['width', 'height'], page.getViewport({scale: 1}));
-        }, pdf.numPages);
-
-        // Set the original sizes for this file.
-        Promise.all(promises).then(setOriginalSizes);
-
-        setTotalPages(pdf.numPages);
-        setUsingFitToScreen(true);
-    };
+    setCurrentPage(previousIndex);
 
-    // Set an effect to process the file resizing when the
-    // viewport sizes changes and it's using "fit to screen".
-    useEffect(() => {
-        // Check if it's using "fit to screen".
-        if (usingFitToScreen) {
-            // Get the "fit to screen" scale.
-            const newScale = originalSizes.map(originalSize =>
-                getFitToScreenScale(viewportSize, originalSize),
-            );
-
-            // Update the scale.
-            setFile(prevValue => R.assoc('scale', newScale, prevValue));
-        }
-    }, [usingFitToScreen, originalSizes, viewportSize]);
+    const previousPage = document.querySelector(
+      `div[data-pdfpage="${previousIndex}"]`,
+    );
 
-    // Reset all the attributes when the file prop changes.
-    useEffect(() => {
-        const file = props.file;
-
-        // if file passed is uploaded file, handle it correctly
-        if (props.file && props.file instanceof File) {
-            file.url = URL.createObjectURL(props.file);
-            file.mimeType = props.file.type;
-        }
-
-        setFile(file);
-        setTotalPages(1);
-        setCurrentPage(0);
-    }, [props.file]);
-
-    return (
-        <div ref={viewportRef} onClick={props.onClick} className="preview-wrapper">
-            <PreviewBar
-                onPageUp={onPageUp}
-                onRotate={onRotate}
-                totalPages={totalPages}
-                onDownload={onDownload}
-                onPageDown={onPageDown}
-                currentPage={currentPage}
-                hidden={props.hideControls}
-            />
-
-            <ViewportContent
-                file={file}
-                containerRef={containerRef}
-                onPageChange={setCurrentPage}
-                onLoadSuccess={onLoadSuccess}
-            />
-
-            <ViewportControl
-                onZoomIn={onZoomIn}
-                onZoomOut={onZoomOut}
-                onFitToScreen={onFitToScreen}
-                hidden={props.hideControls}
-            />
-        </div>
+    // Scroll the viewport to the page position.
+    containerRef.current.scrollTop = previousPage.offsetTop - 10;
+  };
+
+  /**
+   * Handler. Scroll to the next page and update the index.
+   *
+   * @return {void}
+   */
+  const onPageDown = () => {
+    const nextIndex = R.clamp(0, totalPages, currentPage + 1);
+
+    setCurrentPage(nextIndex);
+
+    const nextPage = document.querySelector(`div[data-pdfpage="${nextIndex}"]`);
```

Comment:
Same question about the attribute.

Reply:
Here: https://github.com/vendorpay/react-file-previewer/blob/master/src/PDFViewer.js#L15